File: front-end/src/app/models/livro.ts
import { Autor } from "./autor";
import { Genero } from "./genero";

export interface Livro {
  id: number;
  titulo: string;
  isbn: string;
  autor: Autor;
  genero: Genero;
  sinopse: string;
  capa: string;
}

File: backend/src/main/java/com/litfeed/backend/entity/Livro.java
package com.litfeed.backend.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "Livro")
@Getter
@Setter
public class Livro {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String titulo;
    private String sinopse;
    private String capa;

    @ManyToOne
    @JoinColumn(name = "autor_id", nullable = false)
    private Autor autor;

    @ManyToOne
    @JoinColumn(name = "genero_id", nullable = false)
    private Genero genero;

}
